ui/src/main/java/edu/wpi/grip/ui/MainWindowController.java

```diff
@@ -260,4 +277,43 @@ protected void deploy() {
     dialog.setResizable(true);
     dialog.showAndWait();
   }
+
+  @Subscribe
+  @SuppressWarnings({"PMD.UnusedPrivateMethod", "PMD.UnusedFormalParameter"})
+  private void runStopped(TimerEvent event) {
+    if (!(event.getTarget() instanceof PipelineRunner)) {
+      return;
+    }
+    Platform.runLater(() -> updateElapsedTimeLabel(event.getElapsedTime()));
+  }
+
+  private void updateElapsedTimeLabel(long elapsed) {
+    elapsedTimeLabel.setText(
+        String.format("Ran in %.1f ms (%.1f fps)",
+            elapsed / 1e3,
+            elapsed != 0 ? (1e6 / elapsed) : Double.NaN));
+  }
+
+  @FXML
+  @SuppressWarnings({"PMD.UnusedPrivateMethod"})
+  private void showAnalysis() {
+    Stage analysisStage = new Stage();
+    try {
+      FXMLLoader loader
+          = new FXMLLoader(getClass().getResource("/edu/wpi/grip/ui/analysis/AnalysisWindow.fxml"));
+      analysisStage.setScene(new Scene(loader.load()));
+      AnalysisWindowController controller = loader.getController();
+      controller.setBenchmarker(benchmarkRunner);
+      controller.setStepIndexer(stepIndexer);
+      analysisStage.initOwner(root.getScene().getWindow());
+      analysisStage.setTitle("Pipeline Analysis");
+      analysisStage.getIcons().add(new Image("/edu/wpi/grip/ui/icons/grip.png"));
+      analysisStage.setOnCloseRequest(event -> eventBus.post(BenchmarkEvent.finished()));
+      eventBus.register(controller);
+      analysisStage.showAndWait();
+      eventBus.unregister(controller); // unregister to avoid memory leak
```

Comment:
How does this work on a high DPI screen?
The fonts get all weird on those sort of screens.